drivers/platform/surface/surface_platform_profile.c

```diff
@@ -83,20 +109,43 @@ static int convert_ssam_to_profile(struct ssam_device *sdev, enum ssam_tmp_profi
 	}
 }
 
-static int convert_profile_to_ssam(struct ssam_device *sdev, enum platform_profile_option p)
+
+static int convert_profile_to_ssam_perf(struct ssam_device *sdev, enum platform_profile_option p)
+{
+	switch (p) {
+	case PLATFORM_PROFILE_LOW_POWER:
+		return SSAM_PERF_PROFILE_BATTERY_SAVER;
+
+	case PLATFORM_PROFILE_BALANCED:
+		return SSAM_PERF_PROFILE_NORMAL;
+
+	case PLATFORM_PROFILE_BALANCED_PERFORMANCE:
+		return SSAM_PERF_PROFILE_BETTER_PERFORMANCE;
+
+	case PLATFORM_PROFILE_PERFORMANCE:
+		return SSAM_PERF_PROFILE_BEST_PERFORMANCE;
+
+	default:
+		/* This should have already been caught by platform_profile_store(). */
+		WARN(true, "unsupported platform profile");
+		return -EOPNOTSUPP;
```

Comment:
I'm not sure if `EOPNOTSUPP` is the best value here. Maybe `EINVAL` fits better?

Reply:
Hmm, the current [`convert_profile_to_ssam`](https://github.com/linux-surface/kernel/blob/8cd23b1bb3a8b7a3ef7cec2c37e7e46e6397a858/drivers/platform/surface/surface_platform_profile.c#L104) function uses `EOPNOTSUPP`, so I foolowed that, I'm fine with changing it over, but it feels like they should both return the same error.

Reply:
Oh hmm, you're right. Then we should stick with `EOPNOTSUPP`.